common/txmgr/resender.go

```diff
@@ -102,9 +102,9 @@ func NewResender[
 }
 
 // Start is a comment which satisfies the linter
-func (er *Resender[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Start() {
+func (er *Resender[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Start(ctx context.Context) {
 	er.logger.Debugf("Enabled with poll interval of %s and age threshold of %s", er.interval, er.txConfig.ResendAfterThreshold())
-	go er.runLoop()
+	go er.runLoop(ctx)
```

Comment:
The start context is not meant to be used after start has returned. The background routines should be wired to Close via a stop chan or context derived from a stop chan

Reply:
The context associated with Resender is created as a context.Background(), hmm

Reply:
Sometimes that has to happen, but usually not explicitly - just via the stopChan.NewCtx() helper or something like that